docs/concepts.rst

```diff
@@ -27,40 +32,77 @@ Projects
 
 When you work with CumulusCI, you do so inside a project. A project is a version control repository that contains Salesforce metadata as well as the CumulusCI automation that builds and releases the project. A project usually has a one-to-one relationship with a managed package, if building a package, or with an org implementation. A project may constitute the entirety of a product in the Product Delivery Model, or may be one of multiple projects making up a product.
```

Comment:
Vital to discuss:

1) A concern with focusing on the word project is that when we talk CumulusCI, particularly in the trailheads, we most often discuss packages. Here it sounds like Project is the feted term for what gets deployed . . . but should that be the case? Would it be better to keep package front and center for our terminology? ("A package cannot be deployed without being nested inside a project, which contains SF metadata as well as CCI automation... A project does not need a package, but a project itself is what allows packages to be deployed. It's the elves cobbling away at night for the shoemaker. It's the magic inside the toaster that turns bread into toast.")

2.) Parker and Reed established that a version control repository shouldn't need much explanation to the intended audience, but at some point we need to mention that GitHub _is_ currently the best (only?) version control repository for CCI. (Perhaps "Getting Started"?)

3. Org implementation ("an unmanaged/unlocked package that goes into a specific org; not for universal application (non-proprietary)") needs more detail because I imagine most people using CumulusCI will require the use of at least one per project.